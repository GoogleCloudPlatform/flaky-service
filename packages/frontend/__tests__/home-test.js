// Copyright 2020 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
/* global test, expect */

import React from 'react';
import App from '../src/app';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import nock from 'nock';

const DELAY = 25;
nock.disableNetConnect();

test('renders login button if authorize URL can be fetched', done => {
  const request = nock('http://127.0.0.1:4000')
    .get('/login/github')
    .reply(200, {
      authorization_url: 'http://www.github.com//login/oauth/authorize'
    });
  const component = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <App/>
    </MemoryRouter>
  );
  setTimeout(() => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    request.done();
    return done();
  }, DELAY);
});
