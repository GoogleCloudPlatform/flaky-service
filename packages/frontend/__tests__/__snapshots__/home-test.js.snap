// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login button if authorize URL can be fetched 1`] = `
<div>
  <div
    className="home container-fluid"
  >
    <div
      className="header row"
    >
      <div
        className="col-md-10"
      >
        flaky.dev
      </div>
      <div
        className="col"
      >
        <a
          className="btn btn-primary btn-block"
          href="http://www.github.com//login/oauth/authorize"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Login with GitHub
        </a>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-6 offset-md-2"
      >
        <h1>
          Stop wrestling with flaky tests
        </h1>
        <p>
          flaky.dev collects information from your tests runs, performs statistical analysis, and helps you keep on top of failing tests.
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-md-3"
      />
      <div
        className="col-md-6"
      >
        <a
          className="btn btn-primary btn-block"
          href="http://www.github.com//login/oauth/authorize"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          Create an Account
        </a>
      </div>
    </div>
  </div>
</div>
`;
